commons/src/lib.rs

```diff
@@ -91,16 +91,34 @@ pub fn ensure_query_params(
     Ok(())
 }
 
-/// Make sure client requested a valid content type.
-pub fn ensure_content_type(
+/// Make sure the client can accept the provided media type.
+pub fn validate_content_type(
     headers: &HeaderMap,
     content_type: &'static str,
 ) -> Result<(), GraphError> {
-    let content_json = header::HeaderValue::from_static(content_type);
+    if !headers.contains_key(header::ACCEPT) {
+        return Ok(());
+    }
+
+    let full_type = header::HeaderValue::from_static(content_type);
+    let wildcard = header::HeaderValue::from_static("*");
+    let double_wildcard = header::HeaderValue::from_static("*/*");
+    let top_type = content_type.split("/").next().unwrap_or("");
```

Comment:
Not sure if I understand - if we pass `application/foo` it would still match as the top type matches? Why would we want that?

Reply:
`top_type` isn't used directly, it just feeds `top_type_wildcard`.  If some crazy person sends `Accept: application/*`, then we can serve them `application/json`.  If they send `Accept: application/foo`, we should 400 them.